aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,218 @@
+        if plot_num in range(23) and plot_num not in [14, 15, 16]: # Must be integer in range 0-22, but not 14-16:
+            parameters['INPUTPP']['plot_num'] = int(plot_num) # If this test passes, we can safely cast to int
+        else:
+            raise exceptions.InputValidationError("'plot_num' must be an integer in the range 0-23")
+
+        # Check for valid plot dimension:
+        if dimension in range(5): # Must be in range 0-4:
+            parameters['PLOT']['iflag'] = int(dimension)
+        else:
+            raise exceptions.InputValidationError("'iflag' must be an integer in the range 0-4")
+
+        # Restrict the plot output to the file types that we want to be able to parse
+        if dimension in [0, 1, 4]:
+            parameters['PLOT']['output_format'] = 0    # Gnuplot, 1D
+        elif dimension == 2:
+            parameters['PLOT']['output_format'] = 7    # Gnuplot, 2D
+        elif dimension == 3:
+            parameters['PLOT']['output_format'] = 6    # Gaussian cube, 3D
+        else:
+            raise exceptions.InputValidationError(
+                "Unsupported value, {}, for 'iflag'. Must be in the range 0 to 4".format(dimension)
+            )
+
+        namelists_toprint = self._default_namelists
+
+        input_filename = self.inputs.metadata.options.input_filename
+        with folder.open(input_filename, 'w') as infile:
+            for namelist_name in namelists_toprint:
+                infile.write(u'&{0}\n'.format(namelist_name))
+                # namelist content; set to {} if not present, so that we leave an empty namelist
+                namelist = parameters.pop(namelist_name, {})
+                for key, value in sorted(six.iteritems(namelist)):
+                    infile.write(convert_input_to_namelist_entry(key, value))
+                infile.write(u'/\n')
+
+
+        # Check for specified namelists that are not expected
+        if parameters:
+            raise exceptions.InputValidationError(
+                'The following namelists are specified in parameters, but are '
+                'not valid namelists for the current type of calculation: '
+                '{}'.format(','.join(list(parameters.keys()))))
+
+        remote_copy_list = []
+        local_copy_list = []
+
+        # Copy remote output dir
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        if isinstance(parent_calc_folder, orm.RemoteData):
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._INPUT_SUBFOLDER),
+                self._OUTPUT_SUBFOLDER
+            ))
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._PSEUDO_SUBFOLDER),
+                self._PSEUDO_SUBFOLDER
+            ))
+        elif isinstance(parent_calc_folder, orm.FolderData):
```

Comment:
This is just an FYI: this solution for `FolderData` nodes is not comprehensive I am afraid, because it does not deal with nested folders. It is fine to leave this as is, because the required solution is messy and else where in the plugin these cases are also not dealt with correctly. I also opened a feature request on `aiida-core` to improve this use-case.